File: sandbox-runner/patterns/cancellation-propagation/python/worker.py
import asyncio

from temporalio.client import Client
from temporalio.worker import Worker

from activities import apply_step, compensate_step, hold_reservation
from shared import TASK_QUEUE
from workflows import FulfillmentStep, FulfillOrderWorkflow


async def main() -> None:
    client = await Client.connect("localhost:7233")
    worker = Worker(
        client,
        task_queue=TASK_QUEUE,
        workflows=[FulfillOrderWorkflow, FulfillmentStep],
        activities=[apply_step, hold_reservation, compensate_step],
    )
    print(f"Worker listening on task queue '{TASK_QUEUE}'", flush=True)
    await worker.run()


if __name__ == "__main__":
    asyncio.run(main())

File: sandbox-runner/patterns/cancellation-propagation/typescript/worker.ts
import { Worker } from "@temporalio/worker";

import * as activities from "./activities";
import { TASK_QUEUE } from "./shared";

async function main(): Promise<void> {
  const worker = await Worker.create({
    workflowsPath: require.resolve("./workflows"),
    activities,
    taskQueue: TASK_QUEUE,
  });
  console.log(`Worker listening on task queue '${TASK_QUEUE}'`);
  await worker.run();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
